### DISCUSSION TITLE
How to catch all Exception in one exception_handler?

### DISCUSSION THREAD

**Discussion by axiangcoding** [fastapi/fastapi#10452] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()

@app.exception_handler(Exception)
async def common_exception_handler(request: Request, exc: Exception):
    logger.exception(exc)
    return JSONResponse(
        status_code=500,
        content={
            "message": f"Internal Error",
            "detail": f"{exc}"
        },
    )
```

### Description

Unable to catch subclasses of Exception

### Operating System

Linux, Windows

### FastAPI Version

0.103.2

### Pydantic Version

1.10.13

### Python Version

Python 3.11.5

**Comment by Kludex** [comment]
What's the exception not being caught?

**Reply by axiangcoding** [reply]
I wish to catch all exceptions, their base class is Exception, like a global exception handler.

**Reply by Kludex** [reply]
I understand. What is not working?

**Reply by axiangcoding** [reply]
When an error occurs like ValueError (subclass of Exception), I cannot capture the error and handle it in exception_handler. The error will be handled by Uvicorn server.

Should i provide a minimal demo to describe my problem?

**Reply by n8sty** [reply]
Please do

**Comment by axiangcoding** [comment, marked as answer]
I just notice that when i config `debug=True`, stacktrace will be print into reponse body instead of use exception_handler to format exception message...

**Reply by axiangcoding** [reply]
@Kludex Thanks for help :)
